### ISSUE TITLE
Using arrays as parameters

### ISSUE THREAD

**Issue by mchlr** [gpujs/gpu.js#51]
Hey there guys,

I'm currently working on shifting big calculations from the CPU to the GPU using GPU.js to increase the performance of my app.

I tried the following:
I created a gpu Kernel which should multiplicate each element from ArrayA, with the corresponding Item from ArrayB:
```
var gpuMultiplication = gpu.createKernel(function (ArrayA, ArrayB) {
            return ArrayA[this.thread.x] * ArrayB[this.thread.x];
        }).dimensions([50]);
```

this Code works just fine but only when I'm calling the function using newly created arrays as the parameters. E.g.: `gpuMultiplication([1 ,2, 3, 4, ...], [..., 4, 3, 2, 1]);`

Once I'm trying to pass an array as a variable into the function (`gpuMultiplication(ArrayX, ArrayY);`), gpu.js crashes with the following exception: 

> gpu.js:44 An error occurred compiling the shaders: 
> ERROR: 0:135: 'user_ArrayA' : undeclared identifier 
> ERROR: 0:135: 'user_ArrayASize' : undeclared identifier 
> ERROR: 0:135: 'user_ArrayASize' :  left of '[' is not of type array, matrix, or vector  
> ERROR: 0:135: 'user_ArrayASize' :  left of '[' is not of type array, matrix, or vector
> ERROR: 0:135: 'user_ArrayADim' : undeclared identifier 
> ERROR: 0:135: 'user_ArrayADim' :  left of '[' is not of type array, matrix, or vector  
> ERROR: 0:135: 'user_ArrayADim' :  left of '[' is not of type array, matrix, or vector  
> ERROR: 0:135: 'user_ArrayADim' :  left of '[' is not of type array, matrix, or vector  
> ERROR: 0:135: 'get' : no matching overloaded function found 
> ERROR: 0:135: 'user_ArrayB' : undeclared identifier 
> ERROR: 0:135: 'user_ArrayBSize' : undeclared identifier 
> ERROR: 0:135: 'user_ArrayBSize' :  left of '[' is not of type array, matrix, or vector  
> ERROR: 0:135: 'user_ArrayBSize' :  left of '[' is not of type array, matrix, or vector  
> ERROR: 0:135: 'user_ArrayBDim' : undeclared identifier 
> ERROR: 0:135: 'user_ArrayBDim' :  left of '[' is not of type array, matrix, or vector  
> ERROR: 0:135: 'user_ArrayBDim' :  left of '[' is not of type array, matrix, or vector  
> ERROR: 0:135: 'user_ArrayBDim' :  left of '[' is not of type array, matrix, or vector  
> ERROR: 0:135: 'get' : no matching overloaded function found

Both Arrays are initialized as Float32Array(50) and filled with random values. (Values are only numeric and without any commas)

Am I doing something wrong? Or is it not possible to input arrays like that?
I created my code according to the following example from your website: http://gpu.rocks/examples/

Regards, 
Michael

**Comment by fuzzie360** [comment]
Hi there Michael. Sorry I've been very busy lately. I think I know what is going wrong here, it looks like gpu.js is not detecting Float32Array as an array type. I will fix this soon. Thank you!

**Comment by chonderwild** [comment]
I'm also looking forward to this fix! GPU.JS is such a promising project. Thank you guys!

**Comment by fuzzie360** [comment]
Just fixed it as of hash f9b66e4

Thanks for reporting this bug!

Here's a test you can run on the console of the front page.

```js
var arr = new Float32Array(2);
arr[0] = 1;
arr[1] = 2;

var f = gpu.createKernel(function (arr) {
    return arr[this.thread.x];
}).dimensions([2]);

f(arr) // [1,2]
```
